packages/ssm-cache/index.ts

```diff
@@ -14,9 +14,19 @@ export type SsmCache = {
 
 export const ssmCache: SsmCache = { values: {} };
 
+export class SsmParameterNotFound extends Error {
+  constructor(message: string) {
+    super(message);
+
+    // see: https://github.com/microsoft/TypeScript/wiki/FAQ#why-doesnt-extending-built-ins-like-error-array-and-map-work
+    Object.setPrototypeOf(this, SsmParameterNotFound.prototype);
+    this.name = 'SsmParameterNotFound';
+  }
+}
+
```

Comment:
One thing I'd like to know is why @murilovmachado [did this](https://github.com/techmatters/infrastructure-config/blob/a33835ccf4dc4c3a97e07f0dc0afe2c65ddfed91/lambdas/src/lineSetup/invalidInputPayloadException.ts#L12) in a very similar situation. Could this "plain class" bites us in the future if we are not careful?

Reply:
I *think* it was probably related to the typescript targeting es5 issue I mentioned [here](https://github.com/techmatters/flex-plugins/pull/1009#discussion_r1013981220) and in the ‘see:’ comment.

It shouldn't technically be a problem in HRM, I believe, because we are targeting ES2016. But I added the recommended hack around the issue anyway.

Reply:
Direct link to the typescript FAQ: https://github.com/microsoft/TypeScript/wiki/FAQ#why-doesnt-extending-built-ins-like-error-array-and-map-work